tools/setup/emoji/generate_emoji_listing

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env python3
+import os
+import ujson
+
+from typing import Any, Dict, List
+
+from emoji_setup_utils import emoji_is_universal, EMOJISETS
+from zulip_emoji_map import ZULIP_EMOJI_MAP
+
+TOOLS_DIR = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+ZULIP_PATH = os.path.dirname(TOOLS_DIR)
+EMOJI_MAP_FILE = os.path.join(TOOLS_DIR, 'setup', 'emoji', 'emoji_map.json')
+UNIFIED_REACTIONS_FILE = os.path.join(ZULIP_PATH, 'zerver', 'management', 'data', 'unified_reactions.json')
+EMOJI_DATA_FILE = os.path.join(ZULIP_PATH, 'node_modules', 'emoji-datasource', 'emoji.json')
+EMOJI_CACHE = os.path.join(ZULIP_PATH, 'static', 'generated', 'emoji')
+OUTPUT_FILE = os.path.join(EMOJI_CACHE, 'emoji_listing.html')
+
+with open(EMOJI_DATA_FILE) as fp:
+    EMOJI_DATA = ujson.load(fp)
+with open(UNIFIED_REACTIONS_FILE) as fp:
+    UNIFIED_REACTIONS_MAP = ujson.load(fp)
+with open(EMOJI_MAP_FILE) as fp:
+    EMOJI_MAP = ujson.load(fp)
+
+EMOJI_IMAGE_TEMPLATE = """
+    <div class="emoji emoji-%(emoji_code)s %(emojiset)s" title=%(emojiset)s></div>
+"""
+
+TABLE_ROW_TEMPLATE = """
+    <tr>
+        <td class="new-sorting-info">%(sorting_info)s</td>
+        <td class="emoji-code">%(emoji_code)s</td>
+        <td class="emoji-images">%(images_html)s</td>
+        <td class="zulip-emoji-names">%(zulip_names)s</td>
+        <td class="iamcal-emoji-names">%(iamcal_names)s</td>
+        <td class="gemoji-emoji-names">%(gemoji_names)s</td>
+        <td class="unicode-name">%(unicode_name)s</td>
+    </tr>
+"""
+
+EMOJI_LISTING_TEMPLATE = """
+<html>
+    <head>
+        <link rel = "stylesheet" type = "text/css" href = "/static/generated/emoji/google_sprite.css" />
+        <style>
+            %(table_css)s
+        </style>
+        <title>Zulip emoji listing</title>
+    </head>
+    <body>
+        <table>
+            <thead>
+                <tr>
+                    <th class="new-sorting-info">Sorting info</th>
```

Comment:
Lets make the titles: Category, Codepoint, Images, Zulip, Iamcal (Slack), Gemoji (unordered), Unicode

Reply:
Changed the title except: `Codepoint`. I think `Emoji code` is more appropriate?

Reply:
sounds good